### PR TITLE
CriteriaForm doesn't know the current criteria list by change them

### LINKED ISSUE #2880: CriteriaForm doesn't know the current criteria list by change them

### PR DESCRIPTION
**Pull request by LamaEats**
## PR includes

- [x] Bug Fix

## Related issues

Resolve #2880

### CODE REVIEW

File: src/components/GoalCriteriaSuggest.tsx [hunk 1]
```diff
@@ -157,3 +133,79 @@ export const GoalCriteriaSuggest: React.FC<GoalCriteriaComboBoxProps> = ({
...
+        },
+        [goalId, onGoalCriteriaAdd],
+    );
+
+    const validityData = useMemo(() => {
+        return data.reduce(
+            (acc, item) => {
+                acc.title.push(item.title);
+                acc.sumOfCriteria += Number(item.weight);
+
```
Comment (asabotovich) [inline src/components/GoalCriteriaSuggest.tsx:188]: can we keep it in helper?

File: src/hooks/useGoalResource.ts [hunk 2]
```diff
@@ -287,6 +287,8 @@ export const useGoalResource = (fields: GoalFields, config?: Configuration) => {
         async (val: RemoveCriteriaSchema, invalidateKey?: RefetchKeys | RefetchKeys[]) => {
             await notifyPromise(removeGoalCriteria.mutateAsync(val), 'criteriaDelete');
 
+            // eslint-disable-next-line no-debugger
+            debugger;
```
Comment (asabotovich) [inline src/hooks/useGoalResource.ts:291]: 😱

**Review by asabotovich — CHANGES_REQUESTED** [review]

File: src/hooks/useGoalResource.ts (hunk 2 above)
Comment (LamaEats) [reply src/hooks/useGoalResource.ts:291]: Oh sh

File: src/components/GoalCriteriaSuggest.tsx (hunk 1 above)
Comment (LamaEats) [reply src/components/GoalCriteriaSuggest.tsx:188]: make as a hook

**Review by asabotovich — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**